CI: prerelease_deps failures

My issue is about ...

#### Reproducing code example:
See CI failures in e.g. gh-4933, gh-13795, gh-13699
scipy.scipy (pre_release_deps_source_dist) 
scipy.scipy (prerelease_deps_64bit_blas)

#### Error message:
```
2021-04-04T07:24:38.5930990Z =================================== FAILURES ===================================
2021-04-04T07:24:38.5931727Z _____ TestNdimageInterpolation.test_geometric_transform_with_string_output _____
2021-04-04T07:24:38.5932929Z [gw1] linux -- Python 3.7.10 /opt/hostedtoolcache/Python/3.7.10/x64/bin/python
2021-04-04T07:24:38.5933621Z scipy/ndimage/tests/test_interpolation.py:469: in test_geometric_transform_with_string_output
2021-04-04T07:24:38.5934440Z     out = ndimage.geometric_transform(data, mapping, output='f')
2021-04-04T07:24:38.5935187Z         data       = array([1])
2021-04-04T07:24:38.5935814Z         mapping    = <function TestNdimageInterpolation.test_geometric_transform_with_string_output.<locals>.mapping at 0x7f626ac94050>
2021-04-04T07:24:38.5936540Z         self       = <scipy.ndimage.tests.test_interpolation.TestNdimageInterpolation object at 0x7f626ac81510>
2021-04-04T07:24:38.5937182Z scipy/ndimage/interpolation.py:330: in geometric_transform
2021-04-04T07:24:38.5937725Z     complex_output=complex_output)
2021-04-04T07:24:38.5938210Z         complex_output = False
2021-04-04T07:24:38.5938695Z         cval       = 0.0
2021-04-04T07:24:38.5939165Z         extra_arguments = ()
2021-04-04T07:24:38.5939688Z         extra_keywords = {}
2021-04-04T07:24:38.5940174Z         input      = array([1])
2021-04-04T07:24:38.5940779Z         mapping    = <function TestNdimageInterpolation.test_geometric_transform_with_string_output.<locals>.mapping at 0x7f626ac94050>
2021-04-04T07:24:38.5941579Z         mode       = 'constant'
2021-04-04T07:24:38.5942105Z         order      = 3
2021-04-04T07:24:38.5942788Z         output     = 'f'
2021-04-04T07:24:38.5943482Z         output_shape = (1,)
2021-04-04T07:24:38.5943957Z         prefilter  = True
2021-04-04T07:24:38.5944482Z scipy/ndimage/_ni_support.py:89: in _get_output
2021-04-04T07:24:38.5945011Z     output = numpy.typeDict[output]
2021-04-04T07:24:38.5945666Z         complex_output = False
2021-04-04T07:24:38.5946206Z         input      = array([1])
2021-04-04T07:24:38.5946845Z         output     = 'f'
2021-04-04T07:24:38.5947349Z         shape      = (1,)
2021-04-04T07:24:38.5948116Z /opt/hostedtoolcache/Python/3.7.10/x64/lib/python3.7/site-packages/numpy/__init__.py:297: in __getattr__
2021-04-04T07:24:38.5948764Z     warnings.warn(msg, DeprecationWarning, stacklevel=2)
2021-04-04T07:24:38.5949363Z E   DeprecationWarning: `np.typeDict` is a deprecated alias for `np.sctypeDict`.
2021-04-04T07:24:38.5950043Z         attr       = 'typeDict'
2021-04-04T07:24:38.5950795Z         msg        = '`np.typeDict` is a deprecated alias for `np.sctypeDict`.'
2021-04-04T07:24:38.5951751Z         val        = {'?': <class 'numpy.bool_'>, 0: <class 'numpy.bool_'>, 'byte': <class 'numpy.int8'>, 'b': <class 'numpy.int8'>, ...}
2021-04-04T07:24:38.6618436Z _______ TestNdimageInterpolation.test_map_coordinates_with_string_output _______
2021-04-04T07:24:38.6622966Z [gw1] linux -- Python 3.7.10 /opt/hostedtoolcache/Python/3.7.10/x64/bin/python
2021-04-04T07:24:38.6624921Z scipy/ndimage/tests/test_interpolation.py:544: in test_map_coordinates_with_string_output
2021-04-04T07:24:38.6626109Z     out = ndimage.map_coordinates(data, idx, output='f')
2021-04-04T07:24:38.6626656Z         data       = array([[1]])
2021-04-04T07:24:38.6627093Z         idx        = array([[[0]],
2021-04-04T07:24:38.6627381Z 
2021-04-04T07:24:38.6627724Z        [[0]]])
2021-04-04T07:24:38.6628263Z         self       = <scipy.ndimage.tests.test_interpolation.TestNdimageInterpolation object at 0x7f626ac90c90>
2021-04-04T07:24:38.6628883Z scipy/ndimage/interpolation.py:440: in map_coordinates
2021-04-04T07:24:38.6629378Z     complex_output=complex_output)
2021-04-04T07:24:38.6629822Z         complex_output = False
2021-04-04T07:24:38.6630697Z         coordinates = array([[[0]],
2021-04-04T07:24:38.6630971Z 
2021-04-04T07:24:38.6631328Z        [[0]]])
2021-04-04T07:24:38.6631725Z         cval       = 0.0
2021-04-04T07:24:38.6632140Z         input      = array([[1]])
2021-04-04T07:24:38.6632801Z         mode       = 'constant'
2021-04-04T07:24:38.6633231Z         order      = 3
2021-04-04T07:24:38.6633909Z         output     = 'f'
2021-04-04T07:24:38.6634345Z         output_shape = (1, 1)
2021-04-04T07:24:38.6635162Z         prefilter  = True
2021-04-04T07:24:38.6635635Z scipy/ndimage/_ni_support.py:89: in _get_output
2021-04-04T07:24:38.6636115Z     output = numpy.typeDict[output]
2021-04-04T07:24:38.6636540Z         complex_output = False
2021-04-04T07:24:38.6636979Z         input      = array([[1]])
2021-04-04T07:24:38.6637785Z         output     = 'f'
2021-04-04T07:24:38.6638212Z         shape      = (1, 1)
2021-04-04T07:24:38.6639009Z /opt/hostedtoolcache/Python/3.7.10/x64/lib/python3.7/site-packages/numpy/__init__.py:297: in __getattr__
2021-04-04T07:24:38.6639651Z     warnings.warn(msg, DeprecationWarning, stacklevel=2)
2021-04-04T07:24:38.6640215Z E   DeprecationWarning: `np.typeDict` is a deprecated alias for `np.sctypeDict`.
2021-04-04T07:24:38.6641280Z         attr       = 'typeDict'
2021-04-04T07:24:38.6642110Z         msg        = '`np.typeDict` is a deprecated alias for `np.sctypeDict`.'
2021-04-04T07:24:38.6643133Z         val        = {'?': <class 'numpy.bool_'>, 0: <class 'numpy.bool_'>, 'byte': <class 'numpy.int8'>, 'b': <class 'numpy.int8'>, ...}
2021-04-04T07:24:38.7233429Z ______ TestNdimageInterpolation.test_affine_transform_with_string_output _______
2021-04-04T07:24:38.7235157Z [gw1] linux -- Python 3.7.10 /opt/hostedtoolcache/Python/3.7.10/x64/bin/python
2021-04-04T07:24:38.7235944Z scipy/ndimage/tests/test_interpolation.py:872: in test_affine_transform_with_string_output
2021-04-04T07:24:38.7236788Z     out = ndimage.affine_transform(data, [[1]], output='f')
2021-04-04T07:24:38.7237443Z         data       = array([1])
2021-04-04T07:24:38.7238012Z         self       = <scipy.ndimage.tests.test_interpolation.TestNdimageInterpolation object at 0x7f626abf6090>
2021-04-04T07:24:38.7238643Z scipy/ndimage/interpolation.py:558: in affine_transform
2021-04-04T07:24:38.7239118Z     complex_output=complex_output)
2021-04-04T07:24:38.7240007Z         complex_output = False
2021-04-04T07:24:38.7240420Z         cval       = 0.0
2021-04-04T07:24:38.7240845Z         input      = array([1])
2021-04-04T07:24:38.7241268Z         matrix     = [[1]]
2021-04-04T07:24:38.7241899Z         mode       = 'constant'
2021-04-04T07:24:38.7242344Z         offset     = 0.0
2021-04-04T07:24:38.7242739Z         order      = 3
2021-04-04T07:24:38.7243393Z         output     = 'f'
2021-04-04T07:24:38.7243816Z         output_shape = (1,)
2021-04-04T07:24:38.7244241Z         prefilter  = True
2021-04-04T07:24:38.7244727Z scipy/ndimage/_ni_support.py:89: in _get_output
2021-04-04T07:24:38.7245209Z     output = numpy.typeDict[output]
2021-04-04T07:24:38.7246069Z         complex_output = False
2021-04-04T07:24:38.7246512Z         input      = array([1])
2021-04-04T07:24:38.7247133Z         output     = 'f'
2021-04-04T07:24:38.7247556Z         shape      = (1,)
2021-04-04T07:24:38.7248352Z /opt/hostedtoolcache/Python/3.7.10/x64/lib/python3.7/site-packages/numpy/__init__.py:297: in __getattr__
2021-04-04T07:24:38.7248964Z     warnings.warn(msg, DeprecationWarning, stacklevel=2)
2021-04-04T07:24:38.7249535Z E   DeprecationWarning: `np.typeDict` is a deprecated alias for `np.sctypeDict`.
2021-04-04T07:24:38.7250575Z         attr       = 'typeDict'
2021-04-04T07:24:38.7251397Z         msg        = '`np.typeDict` is a deprecated alias for `np.sctypeDict`.'
2021-04-04T07:24:38.7252421Z         val        = {'?': <class 'numpy.bool_'>, 0: <class 'numpy.bool_'>, 'byte': <class 'numpy.int8'>, 'b': <class 'numpy.int8'>, ...}
2021-04-04T07:24:38.8915187Z ============================= slowest 10 durations =============================
2021-04-04T07:24:38.8928902Z 8.05s call     build/testenv/lib/python3.7/site-packages/scipy/_lib/tests/test_import_cycles.py::test_modules_importable
2021-04-04T07:24:38.8930470Z 7.26s call     build/testenv/lib/python3.7/site-packages/scipy/stats/tests/test_continuous_basic.py::test_cont_basic[500-200-wrapcauchy-arg104]
2021-04-04T07:24:38.8931849Z 4.25s call     build/testenv/lib/python3.7/site-packages/scipy/spatial/tests/test_kdtree.py::test_kdtree_count_neighbous_multiple_r[KDTree]
2021-04-04T07:24:38.8933168Z 3.92s call     build/testenv/lib/python3.7/site-packages/scipy/optimize/tests/test_lsq_linear.py::TestTRF::test_sparse_bounds
2021-04-04T07:24:38.8934473Z 3.90s call     build/testenv/lib/python3.7/site-packages/scipy/spatial/tests/test_kdtree.py::test_kdtree_count_neighbous_multiple_r[cKDTree]
2021-04-04T07:24:38.8936923Z 3.83s call     build/testenv/lib/python3.7/site-packages/scipy/optimize/tests/test_lsq_linear.py::TestTRF::test_large_rank_deficient
2021-04-04T07:24:38.8938529Z 3.81s call     build/testenv/lib/python3.7/site-packages/scipy/optimize/tests/test_lsq_linear.py::TestBVLS::test_large_rank_deficient
2021-04-04T07:24:38.8940521Z 3.42s call     build/testenv/lib/python3.7/site-packages/scipy/optimize/tests/test_optimize.py::test_cobyla_threadsafe
2021-04-04T07:24:38.8942699Z 3.03s call     build/testenv/lib/python3.7/site-packages/scipy/spatial/tests/test_kdtree.py::test_kdtree_box[cKDTree]
2021-04-04T07:24:38.8943724Z 2.81s call     build/testenv/lib/python3.7/site-packages/scipy/spatial/tests/test_kdtree.py::test_kdtree_box[KDTree]
2021-04-04T07:24:38.8944414Z =========================== short test summary info ============================
2021-04-04T07:24:38.8945083Z FAILED scipy/ndimage/tests/test_interpolation.py::TestNdimageInterpolation::test_geometric_transform_with_string_output
2021-04-04T07:24:38.8946169Z FAILED scipy/ndimage/tests/test_interpolation.py::TestNdimageInterpolation::test_map_coordinates_with_string_output
2021-04-04T07:24:38.8947057Z FAILED scipy/ndimage/tests/test_interpolation.py::TestNdimageInterpolation::test_affine_transform_with_string_output
2021-04-04T07:24:38.8947807Z XPASS scipy/linalg/tests/test_matfuncs.py::TestFractionalMatrixPower::test_singular Too unstable across LAPACKs.
2021-04-04T07:24:38.8948478Z XPASS scipy/ndimage/tests/test_datatypes.py::test_uint64_max runs only on darwin
2021-04-04T07:24:38.8949400Z XPASS scipy/optimize/tests/test_linprog.py::TestLinprogIPSparse::test_bug_6690 Fails with ATLAS, see gh-7877
2021-04-04T07:24:38.8950403Z XPASS scipy/optimize/tests/test_linprog.py::TestLinprogIPSparsePresolve::test_bug_6690 Fails with ATLAS, see gh-7877
2021-04-04T07:24:38.8951135Z XPASS scipy/special/tests/test_basic.py::TestCephes::test_fdtri_mysterious_failure Returns nan on i686.
2021-04-04T07:24:38.9005874Z = 3 failed, 31582 passed, 2058 skipped, 92 xfailed, 5 xpassed in 233.04s (0:03:53) =
2021-04-04T07:24:39.6662831Z *** Looks like some tests failed.
2021-04-04T07:24:39.6738665Z ##[error]Bash exited with code '255'.
2021-04-04T07:24:39.6750217Z ##[section]Finishing: Run tests
```

--- a/scipy/ndimage/_ni_support.py
+++ b/scipy/ndimage/_ni_support.py
@@ -86,7 +86,7 @@ def _get_output(output, input, shape=None, complex_output=False):
             output = numpy.promote_types(output, numpy.complex64)
         output = numpy.zeros(shape, dtype=output)
     elif isinstance(output, str):
-        output = numpy.typeDict[output]
+        output = numpy.sctypeDict[output]
         if complex_output and numpy.dtype(output).kind != 'c':
             raise RuntimeError("output must have complex dtype")
         output = numpy.zeros(shape, dtype=output)

